src/components/Navbar.js

```diff
@@ -66,53 +68,8 @@ class Navbar extends React.Component {
 	};
 
 	handleSave = () => {
-    store.dispatch({ type: CLEAR_SAVE_DIRTY_FLAG_ACTION_TYPE })
-		if (checkSourceFileId(this.props.currentOpenFileId)) {
-			FileStorageSystemClient.doSaveSourceContent(
-				BlockTaggingEditorExtension.editor.value(true),
-				this.props.currentOpenFileId.sourceId,
-			).then(success => {
-				if (success) {
-					this.props.dispatchSetToastAction({
-						message: "Saved source file",
-						severity: TOAST_SEVERITY.SUCCESS,
-						open: true
-					});
-				}
-				else {
-					this.props.dispatchSetToastAction({
-						message: "Failed to save source file",
-						severity: TOAST_SEVERITY.ERROR,
-						open: true
-          });
-          store.dispatch({ type: SET_SAVE_DIRTY_FLAG_ACTION_TYPE });
-				}
-			});
-		} else if (checkViewFileId(this.props.currentOpenFileId) && this.props.currentOpenFileId.viewType !== NO_OPEN_FILE_ID.viewType) {
-			FileStorageSystemClient.doSaveViewSpec(
-				this.props.tagsInView,
-				this.props.currentOpenFileId.sourceId,
-				this.props.currentOpenFileId.viewId,
-				this.props.currentOpenFileId.viewType,
-				false,
-				)
-				.then(() => {
-					this.props.dispatchSetToastAction({
-						message: "View saved",
-						severity: TOAST_SEVERITY.SUCCESS,
-						open: true
-					});
-					store.dispatch({ type: CLEAR_SAVE_DIRTY_FLAG_ACTION_TYPE });
-				})
-				.catch(() => {
-          this.props.dispatchSetToastAction({
-            message: "Failed to save view",
-            severity: TOAST_SEVERITY.ERROR,
-            open: true
-          });
-          store.dispatch({ type: SET_SAVE_DIRTY_FLAG_ACTION_TYPE });
-        });
-		}
+		if (checkSourceFileId(this.props.currentOpenFileId)) { handleSaveSourceContent(); }
```

Comment:
Do you think it's worth moving this `if else` into a function since we repeat it elsewhere. I think it needs to be guarded by some flag anyways so it'll be helpful to keep that logic in one place

Reply:
Agreed!